src/main/java/nl/inl/blacklab/search/Hit.java

```diff
@@ -29,7 +29,7 @@
  * This class has public members for the sake of efficiency; this makes a non-trivial difference
  * when iterating over hundreds of thousands of hits.
  */
-public class Hit implements Comparable<Hit>, Cloneable {
+public final class Hit implements Comparable<Hit>, Cloneable {
```

Comment:
My understanding of best-practices is that immutable classes should be marked final so they cannot be subclassed by subclasses with mutators. It might be overkill in this case though